pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -350,20 +354,18 @@ func (c *Controller) Run(stop <-chan struct{}) {
 		c.initNetworkLookup()
 	}
 
-	go func() {
-		cache.WaitForCacheSync(stop, c.HasSynced)
-		c.queue.Run(stop)
-	}()
+	// start all the informers
+	go c.sharedInformers.Start(stop)
```

Comment:
why do we not wait for sync?

Reply:
The informers built from the informerfactory should be run first, then they can be synced.